# Day 2 EOD: A Data Biography of Toolik Field Station Weather

Date: 09/01/2026

In [1]:
import pandas as pd 
url = 'https://eds-217-essential-python.github.io/data/toolik_weather.csv'
df = pd.read_csv(url)


## Question 1

How many rows and how many columns does the dataset have?

 - There are 21 columns and 11,171 rows in this dataset

In [2]:
df.shape

(11171, 21)

What are the column names? Get them as a Python list

In [3]:
list = df.columns.to_list()
print(list)



['Year', 'Month', 'Date', 'LTER_Site', 'Station', 'Daily_AirTemp_Mean_C', 'Flag_Daily_AirTemp_Mean_C', 'Daily_AirTemp_AbsMax_C', 'Flag_Daily_AirTemp_AbsMax_C', 'Daily_AirTemp_AbsMin_C', 'Flag_Daily_AirTemp_AbsMin_C', 'Daily_Precip_Total_mm', 'Flag_Daily_Precip_Total_mm', 'Daily_windsp_mean_msec', 'FLAG_Daily_windsp_mean_msec', 'Daily_Windspeed_AbsMax_m_s', 'Daily_globalrad_total_jcm2', 'FLAG_Daily_globalrad_total_mjm2', 'Moss', 'Soil20cm', 'Comments']


What is the data type of each column?

    - look

In [4]:
print(df.dtypes)

Year                                 int64
Month                                int64
Date                                 int64
LTER_Site                           object
Station                             object
Daily_AirTemp_Mean_C               float64
Flag_Daily_AirTemp_Mean_C           object
Daily_AirTemp_AbsMax_C             float64
Flag_Daily_AirTemp_AbsMax_C         object
Daily_AirTemp_AbsMin_C             float64
Flag_Daily_AirTemp_AbsMin_C         object
Daily_Precip_Total_mm              float64
Flag_Daily_Precip_Total_mm          object
Daily_windsp_mean_msec             float64
FLAG_Daily_windsp_mean_msec         object
Daily_Windspeed_AbsMax_m_s         float64
Daily_globalrad_total_jcm2         float64
FLAG_Daily_globalrad_total_mjm2     object
Moss                               float64
Soil20cm                           float64
Comments                            object
dtype: object


Yesterday’s practice told you this record runs from June 1988 to December 2018. Do not take that on trust. Use .min() and .max() on the Date column to find the exact first and last day and report those in a new cell.

In [5]:
df.describe()

,Year,Month,Date,Daily_AirTemp_Mean_C,Daily_AirTemp_AbsMax_C,Daily_AirTemp_AbsMin_C,Daily_Precip_Total_mm,Daily_windsp_mean_msec,Daily_Windspeed_AbsMax_m_s,Daily_globalrad_total_jcm2,Moss,Soil20cm
count,11171.000000,11171.000000,1.117100e+04,11171.000000,11001.000000,10967.000000,10751.000000,10345.000000,10325.000000,4118.000000,10333.000000,10351.000000
mean,2003.203384,6.570674,2.003271e+07,-8.196437,-3.447132,-13.295924,0.950953,2.995592,6.321288,1324.307188,-1.471402,-2.055647
std,8.831548,3.443637,8.830795e+04,14.831314,14.805298,14.727318,2.749060,1.553574,2.878432,866.670139,7.354649,5.555551
min,1988.000000,1.000000,1.988060e+07,-55.600000,-53.600000,-57.600000,0.000000,0.000000,0.000000,-2.000000,-24.000000,-21.000000
25%,1996.000000,4.000000,1.996012e+07,-19.700000,-14.200000,-25.600000,0.000000,2.000000,4.300000,591.000000,-7.000000,-6.000000
50%,2003.000000,7.000000,2.003092e+07,-7.100000,-2.900000,-12.300000,0.000000,2.800000,5.900000,1268.000000,-2.000000,-1.000000
75%,2011.000000,10.000000,2.011051e+07,4.400000,8.900000,-0.100000,0.400000,3.600000,7.800000,2026.000000,4.000000,2.000000
max,2018.000000,12.000000,2.018123e+07,20.700000,28.200000,14.900000,64.500000,12.500000,27.000000,3205.000000,19.000000,11.000000


## Question 2

Run .info(). How many columns have fewer non-null values than there are rows?
    
    - 15


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11171 entries, 0 to 11170
Data columns (total 21 columns):
 #   Column                           Non-Null Count  Dtype  
---  ------                           --------------  -----  
 0   Year                             11171 non-null  int64  
 1   Month                            11171 non-null  int64  
 2   Date                             11171 non-null  int64  
 3   LTER_Site                        11171 non-null  object 
 4   Station                          11171 non-null  object 
 5   Daily_AirTemp_Mean_C             11171 non-null  float64
 6   Flag_Daily_AirTemp_Mean_C        1310 non-null   object 
 7   Daily_AirTemp_AbsMax_C           11001 non-null  float64
 8   Flag_Daily_AirTemp_AbsMax_C      1020 non-null   object 
 9   Daily_AirTemp_AbsMin_C           10967 non-null  float64
 10  Flag_Daily_AirTemp_AbsMin_C      1236 non-null   object 
 11  Daily_Precip_Total_mm            10751 non-null  float64
 12  Flag_Daily_Precip_

Use .isnull().sum() to count missing values per column. Which three columns are the emptiest?
    
    - FLAG_Daily_windsp_mean_msec
    - FLAG_Daily_globalrad_total_mjm2
    - Flag_Daily_AirTemp_Mean_C


In [7]:
df.isnull().sum()

Year                                   0
Month                                  0
Date                                   0
LTER_Site                              0
Station                                0
Daily_AirTemp_Mean_C                   0
Flag_Daily_AirTemp_Mean_C           9861
Daily_AirTemp_AbsMax_C               170
Flag_Daily_AirTemp_AbsMax_C        10151
Daily_AirTemp_AbsMin_C               204
Flag_Daily_AirTemp_AbsMin_C         9935
Daily_Precip_Total_mm                420
Flag_Daily_Precip_Total_mm          8002
Daily_windsp_mean_msec               826
FLAG_Daily_windsp_mean_msec        11170
Daily_Windspeed_AbsMax_m_s           846
Daily_globalrad_total_jcm2          7053
FLAG_Daily_globalrad_total_mjm2    11156
Moss                                 838
Soil20cm                             820
Comments                            2247
dtype: int64

One of the flag columns is empty in all but a very small number of rows. Which one, and how many values does it actually have? 
    
    - FLAG_Daily_windsp_mean_msec. It only had 1 recording

## Question 3

Use .value_counts() on LTER_Site and on Station. What does each column tell you?
    
    - each column tells me how many counts are recorded and what they are being recorded as (integers)


In [8]:
df['LTER_Site'].value_counts()

LTER_Site
ARC    11171
Name: count, dtype: int64

In [9]:
df['LTER_Site'].value_counts()

LTER_Site
ARC    11171
Name: count, dtype: int64

Use .value_counts() on Flag_Daily_AirTemp_Mean_C. What values appear, and what do you think they mean?
    
    - For Flag_Daily_AirTemp_Mean_C the only values is E and it happeneds 1310 times. For LTER_Site the only value is ARC and it happens 11,171 times.

In [10]:
df['Flag_Daily_AirTemp_Mean_C'].value_counts()

Flag_Daily_AirTemp_Mean_C
E    1310
Name: count, dtype: int64

Use .value_counts() on the Year column and look at both ends of the result, with .head() and then with .tail(). Do all years have the same number of observations? What would explain a year with fewer?

    - No, all years do not have the same number of observations. A year with fewer could be from there not being enough recordings since some of them were recorded everyday for the whole year. The earleir years tend to have fewer recordings.

In [11]:
df['Year'].value_counts()


Year
1992    366
2000    366
1996    366
2012    366
2016    366
2008    366
2004    366
1998    365
1994    365
1997    365
1995    365
1993    365
1991    365
1990    365
1989    365
2010    365
2002    365
1999    365
2001    365
2006    365
2007    365
2005    365
2003    365
2014    365
2009    365
2013    365
2011    365
2017    365
2015    365
2018    365
1988    214
Name: count, dtype: int64

In [12]:
df.head()

,Year,Month,Date,LTER_Site,Station,Daily_AirTemp_Mean_C,Flag_Daily_AirTemp_Mean_C,Daily_AirTemp_AbsMax_C,Flag_Daily_AirTemp_AbsMax_C,Daily_AirTemp_AbsMin_C,...,Daily_Precip_Total_mm,Flag_Daily_Precip_Total_mm,Daily_windsp_mean_msec,FLAG_Daily_windsp_mean_msec,Daily_Windspeed_AbsMax_m_s,Daily_globalrad_total_jcm2,FLAG_Daily_globalrad_total_mjm2,Moss,Soil20cm,Comments
0,1988,6,19880601,ARC,TLKMAIN,8.4,E,NaN,NaN,NaN,...,0.0,E,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Air temp 1 & 5 meter estimated from regressio...
1,1988,6,19880602,ARC,TLKMAIN,6.0,E,NaN,NaN,NaN,...,0.0,E,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Air temp 1 & 5 meter estimated from regressio...
2,1988,6,19880603,ARC,TLKMAIN,5.8,E,NaN,NaN,NaN,...,0.0,E,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Air temp 1 & 5 meter estimated from regressio...
3,1988,6,19880604,ARC,TLKMAIN,1.8,E,NaN,NaN,NaN,...,0.0,E,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Air temp 1 & 5 meter estimated from regressio...
4,1988,6,19880605,ARC,TLKMAIN,6.8,E,NaN,NaN,NaN,...,2.5,E,NaN,NaN,NaN,NaN,NaN,NaN,NaN,Air temp 1 & 5 meter estimated from regressio...


In [13]:
df.tail()

,Year,Month,Date,LTER_Site,Station,Daily_AirTemp_Mean_C,Flag_Daily_AirTemp_Mean_C,Daily_AirTemp_AbsMax_C,Flag_Daily_AirTemp_AbsMax_C,Daily_AirTemp_AbsMin_C,...,Daily_Precip_Total_mm,Flag_Daily_Precip_Total_mm,Daily_windsp_mean_msec,FLAG_Daily_windsp_mean_msec,Daily_Windspeed_AbsMax_m_s,Daily_globalrad_total_jcm2,FLAG_Daily_globalrad_total_mjm2,Moss,Soil20cm,Comments
11166,2018,12,20181227,ARC,TLKMAIN,-21.2,NaN,-7.7,NaN,-26.7,...,0.0,NaN,1.3,NaN,3.0,NaN,NaN,-3.0,-3.0,NaN
11167,2018,12,20181228,ARC,TLKMAIN,-24.2,NaN,-21.4,NaN,-26.3,...,0.0,NaN,1.1,NaN,2.3,NaN,NaN,-3.0,-4.0,NaN
11168,2018,12,20181229,ARC,TLKMAIN,-21.1,NaN,-16.3,NaN,-25.8,...,0.6,NaN,1.9,NaN,3.9,NaN,NaN,-3.0,-4.0,NaN
11169,2018,12,20181230,ARC,TLKMAIN,-17.3,NaN,-7.9,NaN,-28.0,...,0.0,NaN,4.8,NaN,10.1,NaN,NaN,-3.0,-4.0,NaN
11170,2018,12,20181231,ARC,TLKMAIN,-5.5,NaN,-3.3,NaN,-7.9,...,0.1,NaN,7.0,NaN,10.7,NaN,NaN,-3.0,-4.0,NaN


## Question 4

What is the mean daily air temperature across the whole record? The minimum? The maximum?

    - The min is -55.6 and the max is 20.70


In [14]:
df['Daily_AirTemp_Mean_C'].describe()

count    11171.000000
mean        -8.196437
std         14.831314
min        -55.600000
25%        -19.700000
50%         -7.100000
75%          4.400000
max         20.700000
Name: Daily_AirTemp_Mean_C, dtype: float64

Are the mean, the minimum and the maximum plausible for a site above the Arctic Circle? Say why or why not

    - Yes, the minimum is very believalbe and the max seems to be a little warm. But, during the summer months it can get that warm. GLOBAL WARMING WOOOO.

Pick one other numeric column and report its range. Does anything about it look wrong?

    - The Daily_Precip_Total_mm column is underreported and has affected the readings. It is bascially saying that under half of the year it does not rain. 

In [15]:
df['Daily_Precip_Total_mm'].describe()

count    10751.000000
mean         0.950953
std          2.749060
min          0.000000
25%          0.000000
50%          0.000000
75%          0.400000
max         64.500000
Name: Daily_Precip_Total_mm, dtype: float64

## Question 5

Build a list of the five columns you’d keep if you had to hand this dataset to a colleague who only cared about temperature. Use it to make a smaller DataFrame

In [16]:
temp_col = ['Year', 
'Month', 
'Date', 
'Daily_AirTemp_Mean_C', 
'Flag_Daily_AirTemp_Mean_C', 
'Daily_AirTemp_AbsMax_C', 
'Flag_Daily_AirTemp_AbsMax_C', 
'Daily_AirTemp_AbsMin_C', 
'Flag_Daily_AirTemp_AbsMin_C']
df_new = df[temp_col]

Build a dictionary that renames at least two of those columns to something shorter and clearer. Apply it with .rename(columns=...) and confirm the new names.


In [17]:
new_names = {
    'Daily_AirTemp_Mean_C' : 'Mean_Airtemp',
    'Flag_Daily_AirTemp_Mean_C' : 'Flag_Airtemp'
}

df = df.rename(columns = new_names)

## Question 6

What this dataset is. Source, station, what is measured, at what frequency.
Coverage. How many rows, what time span, how many variables.
Condition. What’s missing and where. Which columns you would and would not trust.
Cautions. Two specific things a person using this data should know before computing anything from it.
Fitness for purpose. Could you use this dataset to describe how Arctic summer temperatures have changed over the record? Say yes or no, and say what you would want to check first.



This dataset is from the Toolik Field Station whihch sits on the North Slope of Alaska, above the Arctic Circle, and is part of the Arctic Long Term Ecological Research (LTER) network. The station has been recording daily weather since the late 1988 to 2018. In this dataset the airtemp, windspeed, and time when it was measured is recorded. The dataset has 11,171 recordings and 21 variables. The columns that are missing a significant amount of data are  FLAG_Daily_windsp_mean_msec, FLAG_Daily_globalrad_total_mjm2, and Flag_Daily_AirTemp_Mean_C. There are more but these are missing over 10,000 recordings. 